marin/execution/executor.py

```diff
@@ -178,12 +185,21 @@ class InputName:
 
     step: ExecutorStep | None
     name: str | None
+    should_skip_parent: bool = False
```

Comment:
Does parent refer to `step`'s parents or `step`?  Is this used iff the parent is a pseudo-dependency?

Haven't the read all of the PR, but not clear why this is on `InputName`

Reply:
ok i clarified